packages/core-transaction-pool/src/service-provider.ts

```diff
@@ -18,42 +21,39 @@ export class ServiceProvider extends Providers.ServiceProvider {
      * @memberof ServiceProvider
      */
     public async register(): Promise<void> {
-        this.app
-            .bind(Container.Identifiers.WalletRepository)
-            .to(PoolWalletRepository)
-            .inSingletonScope()
-            .when(Container.Selectors.anyAncestorOrTargetTaggedFirst("state", "pool"));
-
+        this.app.bind(Container.Identifiers.TransactionPoolCollator).to(Collator);
+        this.app.bind(DynamicFeeMatcher).toSelf();
```

Comment:
We should avoid self-binding classes and always bind to a symbol/name because this makes it difficult for a plugin developers to replace this binding.